stm32: remove zephyr specific IRQ defines

erwango opened the pull request:
Remove zephyr specific IRQ definitions and replace with CMSIS ones.
Even if these should be removed eventually by device tree generated constants, it will save some effort when introducing new STM32 series (F7, F2 atm)

erwango committed:
drivers/exti: stm32: Use CMSIS IRQ defines instead of zephyr

stm32cube SDK provides defines for IRQ line numbers.
It was not possible to use them, since enum where not supported
by IRQ_CONNECT macro.
Use them in order to get rid of zephyr manually coded IRQ lines.
They will later be replaced by device tree definitions when
made available

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
arch/stm32: remove irq definition files

Since they are not used anymore because can be replaced
by CMSIS definition or generated from device tree,
remove IRQ definition file soc_irq.h

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

tagunil commented:
@erwango, there are no CMSIS definitions for STM32F4 EXTI16 to EXTI22. Why should these internally connected lines' handlers be enabled here in the first place?

zhaozhongchen commented:
@tagunil @erwango EXTI16 to EXTI22 are shared with others irq on F4 and F2, there should has a define.

erwango committed:
drivers/dma: stm32: Use CMSIS IRQ defines instead of zephyr

stm32cube SDK provides defines for IRQ line numbers.
It was not possible to use them, since enum where not supported
by IRQ_CONNECT macro.
Use them in order to get rid of zephyr manually coded IRQ lines.
They will later be replaced by device tree definitions when
made available

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented:
How about adding DT support for the DMA instead, and getting IRQs from there.

zhaozhongchen commented on drivers/interrupt_controller/exti_stm32.c:
```diff
@@ -113,56 +113,56 @@ void stm32_exti_enable(int line)
 	}
 #elif CONFIG_SOC_SERIES_STM32F1X
```
How about merge EXTI15..0 for F1, F3, F4? eg:
```
#if CONFIG_SOC_SERIES_STM32F1X || CONFIG_SOC_SERIES_STM32F3X || CONFIG_SOC_SERIES_STM32F4X
if (line >= 5 && line <= 9) {
    irqnum = EXTI9_5_IRQn;
} else if 
```

erwango replied:
Let's get it in tree, then we'll be able to proceed with optimisations.

erwango commented:
> How about adding DT support for the DMA instead, and getting IRQs from there.

@galak , this would require to move DMA to Cube LL first (to ensure we get correct DT bindings). I don't know when it will be done as it's not the first prio. Besides this should be done for exti as well.
Meantime people keep adding STM32XX_IRQ when pushing new series (F2, F7 on going right now)...

Would be nice to get rid of these, since work is ready. Holding it will not give me time to work on DMA, just more work in reviews to know if people manually coded the IRQ right.

tagunil commented:
@erwango, I don't understand how F4 is different from e.g. L0 regarding EXTI >= 16 mapping. Why we enable these IRQs for F4, but not for other families?

erwango commented:
@tagunil , well I think there is no good reason and that has to be reviewed. Which should be easier once this PR is merged
